File: examples/LayaAir/src/proto/MsgScenePlayer.ts
namespace proto {
export class MsgScenePlayer
{
	private _uid: number;
	private _scene_rot_pos: MsgSceneRotPos;


	public Encode(): net.Packet {
		let packet: net.Packet = new net.Packet();
		packet.WriteUint(this._uid);
		packet.WriteBuffer(this._scene_rot_pos.GetBuffer());
		return packet;
	}


	constructor(packet: net.Packet) {
		this._uid = packet.ReadUint();
		this._scene_rot_pos = new MsgSceneRotPos(packet);
	}

	public GetBuffer(): ByteBuffer
	{
		return this.Encode().GetBuffer();
	}


	public get uid(): number { return this._uid; }
	public set uid(value: number) { this._uid = value; }
	public get scene_rot_pos(): MsgSceneRotPos { return this._scene_rot_pos; }
	public set scene_rot_pos(value: MsgSceneRotPos) { this._scene_rot_pos = value; }
}
}

File: data/golang/msg_scene_player.go
package proto

import (
	"packet"
)

type MsgScenePlayer struct {
	Uid                      uint32
	SceneRotPos              *MsgSceneRotPos
}

func (this *MsgScenePlayer) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteUint32(this.Uid)
	pack.WriteBytes(this.SceneRotPos.EncodeMsg())

	return pack.Encode(uint16(0))
}

func (this *MsgScenePlayer) EncodeMsg() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteUint32(this.Uid)
	pack.WriteBytes(this.SceneRotPos.EncodeMsg())

	return pack.ReadBytes()
}

func MsgScenePlayerDecode(pack *packet.Packet) *MsgScenePlayer {
	msgScenePlayer := &MsgScenePlayer{}

	msgScenePlayer.Uid = pack.ReadUint32()
	msgScenePlayer.SceneRotPos = MsgSceneRotPosDecode(pack)
	return msgScenePlayer
}
